opendevin/runtime/od_runtime_client/od_runtime_client.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+import os
+import websockets
+import pexpect
+import json
+import shutil
+from typing import Any
+from websockets.exceptions import ConnectionClosed
+from opendevin.events.serialization import event_to_dict, event_from_dict
+from opendevin.events.observation import Observation
+from opendevin.runtime.plugins import PluginRequirement
+from opendevin.events.action import (
+    Action,
+    CmdRunAction,
+    IPythonRunCellAction,
+)
+from opendevin.events.observation import (
+    CmdOutputObservation,
+    ErrorObservation,
+    Observation,
+    IPythonRunCellObservation
+)
+from opendevin.runtime.plugins import (
+    AgentSkillsRequirement,
+    JupyterRequirement,
+    PluginRequirement,
+)
+
+class RuntimeClient():
+    # This runtime will listen to the websocket
+    # When receive an event, it will run the action and send the observation back to the websocket
+
+    def __init__(self) -> None:
+        self.init_shell()
+        self.init_websocket()
+
+    def init_websocket(self) -> None:
+        server = websockets.serve(self.listen, "0.0.0.0", 8080)
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(server)
+        loop.run_forever()
+    
+    def init_shell(self) -> None:
+        self.shell = pexpect.spawn('/bin/bash', encoding='utf-8')
```

Comment:
TODO: we need to figure a way to support different users. Maybe the runtime cli should be run as root?

Reply:
Yes, this is a future TODO. This problem will influence the permission used when init plugin. Both problem need to be addressed in the future.